### PR TITLE
Whitelist nervehttpd

### PR DESCRIPTION
**Pull request by dichiarafrancesco**

### CODE REVIEW

File: src/fullerite/collector/nerve_httpd.go [hunk 1]
```diff
@@ -102,11 +104,20 @@ func (c *NerveHTTPD) Configure(configMap map[string]interface{}) {
 		c.statusTTL = time.Duration(tmpStatusTTL) * time.Second
 	}
 
+	if val, exists := configMap["servicesWhitelist"]; exists {
+		c.servicesWhitelist = config.GetAsSlice(val)
+	} else {
+		c.log.Warn("Required config does not exist for NerveHTTPD: servicesWhitelist")
```
Comment (koikonom) [inline src/fullerite/collector/nerve_httpd.go]: I would not consider servicesWhitelist a required setting.

File: src/fullerite/collector/nerve_httpd.go [hunk 2]
```diff
@@ -120,10 +126,21 @@ func (c *NerveHTTPD) Collect() {
 	c.log.Debug("Finished parsing Nerve config into ", servicePortMap)
 
 	for port, serviceName := range servicePortMap {
-		if !c.checkIfFailed(serviceName, port) {
-			go c.emitHTTPDMetric(serviceName, port)
+		if c.servicesWhitelist == nil || c.serviceInWhitelist(serviceName) {
```
Comment (gnufied) [inline src/fullerite/collector/nerve_httpd.go]: In certain cases just checking for `nil` if the slice is empty works but it is better to be explicit about it and write:

```
if len(c.serviceWhitelist) == 0 
```

To give you an example -  https://gist.github.com/gnufied/0c8056d8be362fd26a085c740a8b5634

Comment (gnufied) [inline src/fullerite/collector/nerve_httpd.go]: what I am trying to say is - `nil` check works in certain cases but `len` check works always

Comment (dichiarafrancesco) [inline src/fullerite/collector/nerve_httpd.go]: +1

**Comment by gnufied** [comment]
:shipit:

**Pull request merged**